guardian/typerighter
# Add fluentbit logging support, sending logs to ELK

## What does this change?

This send our logs to ELK via FluentBit. This is quite simple now that DevX provide FluentBit in their `cdk-base` AMI role. This PR moves to [an ami that include CDK-base](https://amigo.gutools.co.uk/recipes/editorial-tools-focal-java8-ngrams-ARM-cdk-base) and adds the minimal riff-raff changes required to support FluentBit logging. The [commented guidance in the cdk repo](https://github.com/guardian/cdk/blob/9f79cfab8557144d0edc3a20057e2db5bc465906/src/patterns/ec2-app/base.ts#L48) was helpful.

Logs appear in ELK under [`typerighter-rule-manager` and `typerighter-checker`](https://logs.gutools.co.uk/s/editorial-tools/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!f,value:10000),time:(from:now-3M,to:now))&_a=(columns:!(app,message,log),filters:!(),index:'67604290-baa6-11e9-bea2-633437abb232',interval:auto,query:(language:kuery,query:'app.keyword%3Dtyperighter-rule-manager%20OR%20app.keyword%3Dtyperighter-checker'),sort:!(!('@timestamp',desc)))).

I've deployed to CODE and seen logs for the Rule Manager and Checker:

| Logs for Rule Manager | Logs for Checker |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/34686302/228853364-89ae13d4-cdf8-4802-bc9c-48d072016c46.png) | ![image](https://user-images.githubusercontent.com/34686302/228853332-a2867fa6-e225-478b-8b54-bb9d31de5edf.png) |



## How to test

Deploy to CODE and check [ELK](https://logs.gutools.co.uk/s/editorial-tools/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!f,value:10000),time:(from:now-3M,to:now))&_a=(columns:!(app,message,log),filters:!(),index:'67604290-baa6-11e9-bea2-633437abb232',interval:auto,query:(language:kuery,query:'app.keyword%3Dtyperighter-rule-manager%20OR%20app.keyword%3Dtyperighter-checker'),sort:!(!('@timestamp',desc)))) for Typerighter logs.
